plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptionsGroupedProductSelector.php

```diff
@@ -0,0 +1,326 @@
+<?php
+declare(strict_types=1);
+
+namespace Automattic\WooCommerce\Blocks\BlockTypes;
+
+use Automattic\WooCommerce\Admin\Features\Features;
+use Automattic\WooCommerce\Blocks\Utils\StyleAttributesUtils;
+
+/**
+ * Block type for grouped product selector in add to cart with options.
+ */
+class AddToCartWithOptionsGroupedProductSelector extends AbstractBlock {
+	/**
+	 * Block name.
+	 *
+	 * @var string
+	 */
+	protected $block_name = 'add-to-cart-with-options-grouped-product-selector';
+
+	/**
+	 * Render variation label.
+	 *
+	 * @param string $attribute_name Name of the attribute.
+	 * @return string Rendered label HTML.
+	 */
+	protected function render_variation_label( $attribute_name ): string {
+		$label_id   = esc_attr( 'attribute_' . sanitize_title( $attribute_name ) );
+		$label_text = wc_attribute_label( $attribute_name );
+
+		$html = sprintf(
+			'<label class="wc-block-product-add-to-cart-attribute-label" for="%s">%s</label>',
+			$label_id,
```

Comment:
Nit: I would personally move the `esc_attr()` from L27 into here, so we escape the attribute right before printing it. This way it's cleared that it has been escaped.

```suggestion
			esc_attr( $label_id ),
```